ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTree.java

```diff
@@ -269,15 +276,23 @@ private void updateOptimalLeafWithNextDepthBestLeaf(AbstractNetworkPool networkP
                 }
                 try {
                     if (combinationFulfillingStopCriterion.isEmpty() || deterministicNetworkActionCombinationComparison(naCombination, combinationFulfillingStopCriterion.get()) < 0) {
-                        // Reset range action to their pre-perimeter set-points
-                        // TODO : instead of this parameter, clean rangeActions containted in Leaf. There should only be the optimized state's range actions
-                        // but then other RangeActionActivatinoResult in FARAO initialized with leaf's range actions should be initialized with prePerimeterOutput range Actions + leaf's
-                        // optimized range actions.
-                        input.getOptimizationPerimeter().getRangeActions().forEach(ra ->
-                            ra.apply(networkClone, input.getPrePerimeterResult().getRangeActionSetpointResult().getSetpoint(ra))
-                        );
-
-                        optimizeNextLeafAndUpdate(naCombination, networkClone);
+                        // todo: Probably change the doc here
+                        // Apply range actions that has been changed by the previous leaf on the network to start next depth leaves
+                        // from previous optimal leaf starting point
+                        // TODO: we can wonder if it's better to do this here or at creation of each leaves or at each evaluation/optimization
+
+                        if (naCombinations.get(naCombination)) {
+                            previousDepthOptimalLeaf.getRangeActions().forEach(ra ->
+                                ra.apply(networkClone, input.getPrePerimeterResult().getRangeActionSetpointResult().getSetpoint(ra))
+                            );
+                        } else {
+                            input.getOptimizationPerimeter().getRangeActions()
+                                .forEach(ra ->
+                                        ra.apply(networkClone, previousDepthOptimalLeaf.getOptimizedSetpoint(ra, input.getOptimizationPerimeter().getMainOptimizationState()))
```

Comment:
In this case, the boolean is false which means we want to keep RangeActions applied in the previousDepthOptimalLeaf. 

We want to have the setpoint corresponding to the result of this Leaf (previousDepthOptimalLeaf.getOptimizedSetpoint) and not the PrePerimeterResult one (what you wrote above)

Reply:
sorry, i wrote the comment on the wrong line. I was referring to the point discussed yesterday for which you said there was a specific reason explaining why we couldn't use exactly the same input. .... as previously

Reply:
Oh yes I remember. In fact, the reason I thought we needed this change is not valid anymore :/. 

I can go back the previous behavior but i fail to understand the difference.

 Because before this commit you made in February https://github.com/farao-community/farao-core/pull/728/commits/67b8c668d8625f72cadc159d9ef8c327105ba488 inside your PR "Consider ARAs in 2P", it was  **previousDepthOptimalLeaf**.getRangeActions() and you changed it to **input.getPreOptimizationPerimeter**.getRangeActions()

Reply:
Thanks ! I don't rememeber the context behind this commit but could you check that nothing's broken on cases with a 2A ?

Reply:
I ran every cucumber test tagged 2nd preventive (and also every "fast" tests) and they all pass, whether it's **previousDepthOptimalLeaf**.getRangeActions() or **input.getOptimizationPerimeter**.getRangeActions(). 

To make the code easier to read, i will choose **previousDepthOptimalLeaf** as it's also used in the `if` section right above.